server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -52,7 +52,7 @@ import qualified Hasura.GraphQL.Transport.WebSocket.Server   as WS
 import qualified Hasura.Logging                              as L
 import qualified Hasura.Server.Telemetry.Counters            as Telem
 
-
+-- TODO what are the invariants here? How are they maintained?
```

Comment:
`OperationMap` maintains the state of active subscriptions. When a connection is initialized it is empty, `onStart` and `onStop` can potentially add and remove operations and `onClose` cleans up the connection's operations.